src/tmpfiles/tmpfiles.c

```diff
@@ -3724,6 +3740,14 @@ static int parse_line(
                                 CREATE_SUBVOLUME_INHERIT_QUOTA,
                                 CREATE_SUBVOLUME_NEW_QUOTA) ? 0755 : 0644;
 
+        if (missing_user_or_group && (i.mode & ~0777)) {
+                /* Refuse any special bits for nodes where we couldn't resolve the ownership properly. */
+                mode_t new = i.mode & 0777;
+                log_syntax(NULL, LOG_ERR, fname, line, r,
+                           "Changing mode %o to %o because of changed ownership", i.mode, new);
```

Comment:
probably a good idea to prefix the two `%o` with 0, to indicate they are octal.

also, maybe a full stop at the end of the log msg.